### PR TITLE
TF notebook overhaul

### PR DESCRIPTION
**Pull request by Rocketknight1**
Overhauls all TF notebooks 

- General text fixes and rewrites for clarity
- Compatibility fixes for TF 2.8+
- Metric callbacks and fixes to metric calculations
- Everything tested and rerun

### CODE REVIEW

File: examples/language_modeling-tf.ipynb [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "id": "X4cRE8IbIrIV"
```
Comment (sgugger) [inline examples/language_modeling-tf.ipynb:6]: <p>Leave the part that says you need to uncomment the lines maybe?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/137/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (sgugger) [inline examples/language_modeling-tf.ipynb:6]: <p>There is nothing to uncomment anymore here. I'd leave out the login part as we don't want people to rely on it anymore, it soon won't be possible anymore.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/137/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sgugger) [inline examples/language_modeling-tf.ipynb:6]: <p>I think it needs 4.16.0 minimum now?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/137/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (sgugger) [inline examples/language_modeling-tf.ipynb:6]: <p>I don't know if this the UI or if it's a typo but there seems to be a space in a utomatically.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/137/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (Rocketknight1) [reply examples/language_modeling-tf.ipynb:6]: <p>I believe that's just the UI!</p>

**Review by sgugger — APPROVED** [review]
Great!

**Pull request merged**